processes/record_pipeline.py

```diff
@@ -28,15 +29,22 @@ def __init__(self, *args):
         self.record_clusterer = RecordClusterer(db_manager=self.db_manager)
         self.link_fulfiller = LinkFulfiller(db_manager=self.db_manager)
 
-    def runProcess(self):
+    def runProcess(self, max_attempts: int=4):
         try:
-            while message := self.rabbitmq_manager.getMessageFromQueue(self.record_queue):
-                message_props, _, message_body = message
+             for attempt in range(0, max_attempts):
+                wait_time = 30 * attempt
 
-                if not message_props or not message_body:
-                    break
-                
-                self._process_message(message)
+                if wait_time:
+                    logger.info(f'Waiting {wait_time}s for record messages')
+                    sleep(wait_time)
+
+                while message := self.rabbitmq_manager.getMessageFromQueue(self.record_queue):
+                    message_props, _, message_body = message
+
+                    if not message_props or not message_body:
+                        break
+                    
+                    self._process_message(message)
```

Comment:
Is this loop for retries? Do we need to add a try / except inside the loop body or am I misunderstanding?

Reply:
oh I see, it's for if it doesn't receive a message, carry on!

Reply:
it is yes! we should migrate away from message pollers like this and instead use batch consumption. i was reading this morning that rabbitmq definitely tries to sway people away from 1x1 message pollers lol.